### PR TITLE
external dependency file for automate #5437

### PR DESCRIPTION
**Pull request by shivangi-singh1**
### :nut_and_bolt: Description: What code changed, and why?
Added external dependencies for Automate and A2HA under the dev-docs folder.

### :chains: Related Resources 
Issue Link: https://github.com/chef/automate/issues/5437 

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: dev-docs/external_dependency.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+   * pg-sidecar-service
+     * Version: 9.6.11/20190305213520 
+
+20. sqitch_pg
+   * pg-sidecar-service
+     * Version: 3.7.4/20190305225225
+
+21.  https://github.com/chef/automate/blob/master/go.mod
+
+22. https://github.com/chef/automate/blob/master/vendor.go
```
Comment (vivek-yadav) [inline dev-docs/external_dependency.md:89]: remove this

File: dev-docs/external_dependency.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+19. postgresql-client
+   * pg-sidecar-service
+     * Version: 9.6.11/20190305213520 
+
+20. sqitch_pg
+   * pg-sidecar-service
+     * Version: 3.7.4/20190305225225
+
+21.  https://github.com/chef/automate/blob/master/go.mod
```
Comment (vivek-yadav) [inline dev-docs/external_dependency.md:87]: Please add links with some context and in proper MD link form.

**Review by kalroy — CHANGES_REQUESTED** [review]
I do not see some frequently used habitat packages in Automate.
To name a few:
- glibc
- bash
- mlsa

Please go through the components once more. Looks like we have missed some.

**Comment by shivangi-singh1** [comment]
> I do not see some frequently used habitat packages in Automate.
> To name a few:
> 
> * glibc
> * bash
> * mlsa
> 
> Please go through the components once more. Looks like we have missed some.

Hab dependencies which do not have version specified have not been documented as we do not control their upgrade.

File: dev-docs/external_dependency.md (hunk 2 above)
Comment (shivangi-singh1) [reply dev-docs/external_dependency.md]: done

File: dev-docs/external_dependency.md (hunk 1 above)
Comment (shivangi-singh1) [reply dev-docs/external_dependency.md]: done

**Review by kalroy — APPROVED** [review]

**Comment by kalroy** [comment]
As discussed, the habitat packages may be skipped for the time being.

**Review by vivek-yadav — APPROVED** [review]

**Pull request merged**